### PR TITLE
feat(telecom): add ont shipping address on move and migrate steps

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-585
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

## Description

Add ONT shipping address selection on move and migrate offers.

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      options: {
+        allowCreation: true,
+        allowEdition: false,
+      },
+    };
+    this.loading = {
+      init: false,
+    };
+
+    this.loading.init = true;
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:35]: **nitpick:** 2 statements can be combined
```suggestion
    this.loading = {
      init: true,
    };
```

File: packages/manager/apps/telecom/src/app/telecom/pack/move/ont-shipping/move-ont-shipping.controller.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      },
+    };
+    this.loading = {
+      init: false,
+    };
+    this.ontShipping = {
+      address: null,
+    };
+
+    this.loading.init = true;
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/ont-shipping/move-ont-shipping.controller.js:29]:
```suggestion
    this.loading = {
      init: true,
    };
    this.ontShipping = {
      address: null,
    };
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    this.process.ontShipping = {
+      address: null,
+    };
+
+    // Initialize ONT address shipping
+    return this.OvhApiPackXdsl.v6()
+      .shippingAddresses({
+        packName: this.process.pack.packName,
+        context: 'voipLine',
+      })
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:46]: For avoid to use deprecated library, you can create a function in service file that call directly the api call
```js
this.$http.get(`/pack/xdsl/${packName}/shippingAddresses`).then(({data}) => data);
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                line1: ontShippingAddress.address,
+                city: ontShippingAddress.cityName,
+                country: ontShippingAddress.countryCode,
+                zip: ontShippingAddress.zipCode,
+              },
+              firstName: ontShippingAddress.firstName,
+              lastName: ontShippingAddress.lastName,
+              id: ontShippingAddress.shippingId,
+            }),
+        );
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:62]: You can use the native javascript function for avoid to use lodash library
```suggestion
        this.ovhContactOptions.customList = ontShippingAddresses.map((ontShippingAddress) =>
            new this.OvhContact({
              address: {
                line1: ontShippingAddress.address,
                city: ontShippingAddress.cityName,
                country: ontShippingAddress.countryCode,
                zip: ontShippingAddress.zipCode,
              },
              firstName: ontShippingAddress.firstName,
              lastName: ontShippingAddress.lastName,
              id: ontShippingAddress.shippingId,
            }),
        );
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        this.process.ontShipping.address = head(
+          this.ovhContactOptions.customList,
+        );
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:66]:
```suggestion
        this.process.ontShipping.address = this.ovhContactOptions.customList.at(0);
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      })
+      .catch((error) => {
+        const msgErr = `${this.$translate(
+          'telecom_pack_migration_ont_shipping_addresses_error',
+        )} ${get(error, 'data.message', '')}`;
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:71]: Same no need get lodash
```suggestion
        )} ${error?.data?.message}`;
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  /* =============================
+     =        ACTIONS            =
+     ============================= */
+  nextStep() {
+    this.process.currentStep = 'confirm';
+    if (this.process.selectedOffer.needNewModem) {
+      this.process.currentStep = 'shipping';
+    } else if (this.process.selectedOffer.needMeeting) {
+      this.process.currentStep = 'meeting';
+    }
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:89]: You have to move the constant in constant files.
And if this.process.selectedOffer.needNewModem for example you change twice time yours this.process.currentStep and second this.process.currentStep = 'shipping', maye there is an other best way

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.html [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+                    >
+                    </a>
+                    <button
+                        type="button"
+                        class="btn btn-primary"
+                        data-ng-disabled="!$ctrl.process.ontShipping.address"
+                        data-translate="telecom_pack_migration_ont_shipping_confirm"
+                        data-ng-click="$ctrl.nextStep()"
+                    ></button>
+                </div>
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.html:53]:
```suggestion
                <div data-ng-if="!$ctrl.loading.init">
                    <!-- CONTACT CHOICE -->
                    <div class="contact-choice">
                        <h3
                            data-translate="telecom_pack_migration_ont_shipping_contact"
                            class="mb-3 mt-5"
                        ></h3>

                        <ovh-contact
                            data-ng-model="$ctrl.process.ontShipping.address"
                            data-ovh-contact-choice-options="$ctrl.ovhContactOptions"
                        >
                            <div class="text-center">
                                <oui-spinner></oui-spinner>
                            </div>
                        </ovh-contact>
                    </div>
                    <!-- CONTACT CHOICE -->

                <div class="mt-5">
                    <a
                        class="btn btn-default"
                        data-translate="telecom_pack_migration_confirm_cancel_button"
                        data-ui-sref="telecom.packs.pack"
                    >
                    </a>
                    <button
                        type="button"
                        class="btn btn-primary"
                        data-ng-disabled="!$ctrl.process.ontShipping.address"
                        data-translate="telecom_pack_migration_ont_shipping_confirm"
                        data-ng-click="$ctrl.nextStep()"
                    ></button>
                </div>
             </div>
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/pack-migration.html [hunk 9]
```diff
@@ -33,6 +33,13 @@
         <pack-migration-building-details></pack-migration-building-details>
     </div>
 
+    <!-- ONT SHIPPING STEP -->
+    <div
+        data-ng-if="!$ctrl.loading.init && $ctrl.process.currentStep === 'ontShipping'"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/pack-migration.html:38]:
```suggestion
        data-ng-if="$ctrl.process.currentStep === $ctrl.MY_SUPER_VARIABLE.ontShipping"
```

Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/pack-migration.html:38]:
```suggestion
    <!-- ONT SHIPPING STEP -->
    <div data-ng-if="!$ctrl.loading.init">
    <div
        data-ng-if="$ctrl.process.currentStep === 'ontShipping'"
```
You can use only one div for each step for avoid to duplicate the code data-ng-if="!$ctrl.loading.init" for each step

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/service-delete/pack-migration-service-delete.html [hunk 10]
```diff
@@ -96,7 +96,7 @@
             type="button"
             class="btn btn-primary"
             data-translate="telecom_pack_migration_service_delete_confirm"
-            data-ng-click="$ctrl.process.currentStep = $ctrl.process.selectedOffer.needNewModem ? 'shipping' : $ctrl.process.selectedOffer.needMeeting ? 'meeting' : 'confirm'"
+            data-ng-click="$ctrl.process.currentStep = $ctrl.process.selectedOffer.customOntAddress ? 'ontShipping' : $ctrl.process.selectedOffer.needNewModem ? 'shipping' : $ctrl.process.selectedOffer.needMeeting ? 'meeting' : 'confirm'"
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/pack/migration/service-delete/pack-migration-service-delete.html:99]: And in addition, move also the constant in constant file

**Review by ghyenne — CHANGES_REQUESTED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/index.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+import uiRouter from '@uirouter/angularjs';
+import angularTranslate from 'angular-translate';
+
+import component from './pack-migration-ont-shipping.component';
+
+const moduleName = 'ovhManagerTelecomPackMigrationOntShipping';
+
+angular
+  .module(moduleName, [ngTranslateAsyncLoader, uiRouter, angularTranslate])
+  .component('packMigrationOntShipping', component);
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/index.js]: No, other component names are 'pack-migration-...', if I put 'telecom-pack-migration-ont-shipping', I need to change all component names.

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+              address: {
+                line1: ontShippingAddress.address,
+                city: ontShippingAddress.cityName,
+                country: ontShippingAddress.countryCode,
+                zip: ontShippingAddress.zipCode,
+              },
+              firstName: ontShippingAddress.firstName,
+              lastName: ontShippingAddress.lastName,
+              id: ontShippingAddress.shippingId,
+            }),
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:57]:
```suggestion
          ({address, cityName, countryCode, zipCode, firstName, lastName, shippingId}) =>
            new this.OvhContact({
              address: {
                line1: address,
                city: cityName,
                country: countryCode,
                zip: zipCode,
              },
              firstName: firstName,
              lastName: lastName,
              id: shippingId,
            }),
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/translations/Messages_fr_FR.json [hunk 13]
```diff
@@ -204,5 +204,10 @@
   "telecom_pack_migration_no_meeting_available": "Aucun RDV n'est disponible pour le moment, vous serez recontacté par téléphone pour la planification.",
   "telecom_pack_migration_pto_multi_otp": "J'ai un boîtier et je souhaite la création d'une ligne supplémentaire",
   "telecom_pack_migration_pto_multi_otp_description": "Un technicien procèdera à l'installation d'une seconde prise optique (PTO) lors de son intervention.",
-  "telecom_pack_migration_pto_otp_not_mandatory": "Cette référence PTO est optionnelle dans le cas présent."
+  "telecom_pack_migration_pto_otp_not_mandatory": "Cette référence PTO est optionnelle dans le cas présent.",
+  "telecom_pack_migration_ont_shipping_title": "Choisissez l'adresse de livraison de votre ONT",
+  "telecom_pack_migration_ont_shipping_explain": "L'ONT (Optical Network Termination) est un équipement réseau qui perme de raccorder votre modem à la PTO.",
+  "telecom_pack_migration_ont_shipping_contact": "Contact de livraison",
+  "telecom_pack_migration_ont_shipping_confirm": "Confirmer ce contact de livraison",
+  "telecom_pack_migration_ont_shipping_addresses_error": "Une erreur est survenue lors du chargement de vos contacts."
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/translations/Messages_fr_FR.json:212]:
```suggestion
  "telecom_pack_migration_ont_shipping_addresses_error": "Une erreur est survenue lors du chargement de vos contacts. {{ error }}"
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+          0,
+        );
+      })
+      .catch((error) => {
+        const mgError = this.$translate.instant(
+          'telecom_pack_migration_ont_shipping_addresses_error',
+          { error: error.data?.message },
+        );
+        this.TucToast.error(mgError);
+        return this.$q.reject(error);
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:77]:
```suggestion
```
I don't think is needed you can also remove the import of $q

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 15]
```diff
@@ -0,0 +1,94 @@
+import { PROCESS_STEP, CONTEXT } from '../pack-migration.constant';
+
+export default class PackMigrationOntShippingCtrl {
+  /* @ngInject */
+  constructor(
+    $q,
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:6]:
```suggestion
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+    $translate,
+    TucPackMigrationProcess,
+    OvhContact,
+    PackMigrationOntShippingService,
+    TucToast,
+  ) {
+    this.$q = $q;
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:13]:
```suggestion
```
not used

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+        this.process.ontShipping.address = this.ovhContactOptions.customList.at(
+          0,
+        );
+      })
+      .catch((error) => {
+        const mgError = this.$translate.instant(
+          'telecom_pack_migration_ont_shipping_addresses_error',
+          { error: error.data?.message },
+        );
+        this.TucToast.error(mgError);
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.controller.js:76]:
```suggestion
        this.TucToast.error(this.$translate.instant(
          'telecom_pack_migration_ont_shipping_addresses_error',
          { error: error.data?.message },
        ));
```

**Review by ghyenne — DISMISSED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.service.js [hunk 18]
```diff
@@ -0,0 +1,12 @@
+export default class PackMigrationOntShippingService {
+  /* @nbgInject */
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/ont-shipping/pack-migration-ont-shipping.service.js:2]:
```suggestion
  /* @ngInject */
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by qpavy — APPROVED** [review]

**Review by acavalloni — APPROVED** [review]

**Pull request merged**